### PR TITLE
[BEAM-11665] Publish Beam GCP BOM

### PR DESCRIPTION
**Pull request by kileys**
- Create a Beam GCP BOM for projects that use Beam and Google Cloud Platform.
- Create a Maven archetype using the new BOM and run examples on it
- Add BOM validation to release jenkins test and validation script

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by kileys** [comment]
Run Gradle Publish

**Comment by kileys** [comment]
Run Gradle Publish

**Comment by suztomo** [comment]
@kileys Can I see the generated BOM somewhere?

File: sdks/java/bom/common.gradle [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                        unsubscribe = "dev-unsubscribe@beam.apache.org"
+                        post = "dev@beam.apache.org"
+                        archive = "http://www.mail-archive.com/dev%beam.apache.org"
+                    }
+                    mailingList {
+                        name = "Beam User"
+                        subscribe = "user-subscribe@beam.apache.org"
+                        unsubscribe = "user-unsubscribe@beam.apache.org"
+                        post = "user@beam.apache.org"
+                        archive = "http://www.mail-archive.com/user%beam.apache.org"
```
Comment (elharo) [inline sdks/java/bom/common.gradle:73]: This URL is 404
You might have an issue around the percent sign

File: sdks/java/bom/common.gradle [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                    }
+                    mailingList {
+                        name = "Beam Commits"
+                        subscribe = "commits-subscribe@beam.apache.org"
+                        unsubscribe = "commits-unsubscribe@beam.apache.org"
+                        post = "commits@beam.apache.org"
+                        archive = "http://www.mail-archive.com/commits%beam.apache.org"
```
Comment (elharo) [inline sdks/java/bom/common.gradle:80]: http://www.mail-archive.com/commits%beam.apache.org

File: sdks/java/bom/common.gradle [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                    }
+                }
+                developers {
+                    developer {
+                        name = "The Apache Beam Team"
+                        email = "dev@beam.apache.org"
+                        url = "http://beam.apache.org"
```
Comment (elharo) [inline sdks/java/bom/common.gradle:87]: https here and elsewhere

File: sdks/java/bom/common.gradle (hunk 1 above)
Comment (kileys) [reply sdks/java/bom/common.gradle]: Fixed

**Comment by kileys** [comment]
> @kileys Can I see the generated BOM somewhere?
I updated the artifact name, but you can see the pom here:

https://repository.apache.org/content/repositories/snapshots/org/apache/beam/beam-sdks-java-bom-gcp/2.28.0-SNAPSHOT/beam-sdks-java-bom-gcp-2.28.0-20210112.211655-2.pom

**Comment by kileys** [comment]
Run Gradle Publish

**Review by elharo — COMMENTED** [review]
1. `<url>http://beam.apache.org</url>` should use https
2 `<inceptionYear>2016</inceptionYear>` should be 2021
3. `<organizationUrl>http://www.apache.org</organizationUrl>` should use https
4. `<archive>http://www.mail-archive.com/dev%beam.apache.org</archive>` should use https

**Comment by kileys** [comment]
Run Dataflow PostRelease

**Comment by kileys** [comment]
Run Dataflow PostRelease

**Comment by kileys** [comment]
Run Dataflow PostRelease

**Comment by kileys** [comment]
> 1. `<url>http://beam.apache.org</url>` should use https
>    2 `<inceptionYear>2016</inceptionYear>` should be 2021
> 2. `<organizationUrl>http://www.apache.org</organizationUrl>` should use https
> 3. `<archive>http://www.mail-archive.com/dev%beam.apache.org</archive>` should use https

That's an outdated pom before I made the changes. Also, for #2, it's inception year not current year.

Here's the newest pom:
https://repository.apache.org/content/repositories/snapshots/org/apache/beam/beam-sdks-java-google-cloud-platform-bom/2.28.0-SNAPSHOT/beam-sdks-java-google-cloud-platform-bom-2.28.0-20210120.122741-1.pom

**Comment by kileys** [comment]
Run Dataflow PostRelease

**Comment by kileys** [comment]
R: @kennknowles

**Comment by suztomo** [comment]
I see it's working as expected. (Effective pom after I replaced the version iwth the latest release 2.27: https://gist.github.com/suztomo/ab8bf9dc2653ce2a886a5a57a0204eb2#file-gistfile1-txt-L190 )

I want Guava version to be more higher: https://github.com/apache/beam/pull/13740

**Review by suztomo — APPROVED** [review]

**Review by kennknowles — APPROVED** [review]

File: release/src/main/scripts/run_rc_validation.sh [hunk 4]
```diff
@@ -285,6 +285,17 @@ if [[ ("$java_mobile_game_direct" = true || "$java_mobile_game_dataflow" = true)
     -PbqDataset=${MOBILE_GAME_DATASET} \
     -PpubsubTopic=${MOBILE_GAME_PUBSUB_TOPIC} \
     -PgcsBucket=${USER_GCS_BUCKET:5}  # skip 'gs://' prefix
+
+    echo "**************************************************************************"
+    echo "* Java mobile game on DataflowRunner using Beam GCP BOM: UserScore, HourlyTeamScore, Leaderboard"
```
Comment (udim) [inline release/src/main/scripts/run_rc_validation.sh:290]: Should this be included as part of the release validation spreadsheet?
In other words, is this required to pass to validate the RC?

**Pull request merged**